src/cirrus/builtins/functions/update-state/lambda_function.py

```diff
@@ -167,20 +166,22 @@ def parse_event(event):
             if status == FAILED:
                 error = get_execution_error(event['detail']['executionArn'])
             return (
-                ProcessPayload.from_event(json.loads(event['detail']['input'])),
+                ProcessPayload(json.loads(event['detail']['input'])),
+                ProcessPayload(json.loads(event['detail']['output']))
```

Comment:
since the `output_payload` can be `None`, while the `status == "SUCCEEDED"` you are opening yourself up for a bug` in `workflow_completed` (something like `'NoneType' object has no attribute 'next_payloads'`)

--> check for `None` and return early

Reply:
I thought long and hard about the value of output, and I couldn't see any circumstance where it would be null from a successful workflow execution. I was originally considering a more complicated way of handling that in the `parse_event` function, but decided the complexity was not worth it. That said, with this simpler way of returning both input and output payloads, you're right, it is easy enough to check if it has a value before using it in `workflow_completed`. I've added that check.

`workflow_completed` is so named because the corresponding cirrus state is `COMPLETED`, however I see how that is ambiguous with regard to this function name. I plan a fair bit of refactoring for this function in the near future, at which time I will either rename the function or replace it with entirely different semantics.